src/Plugin/rest/resource/DbResource.php

```diff
@@ -125,6 +126,27 @@ public function post($workspace, ContentEntityInterface $entity = NULL) {
       if (!$field->access('create')) {
         throw new AccessDeniedHttpException(t('Access denied on creating field @field.', array('@field' => $field_name)));
       }
+
+      // Save the files for file and image fields.
+      if ($field instanceof FileFieldItemList) {
+        foreach ($field as $delta => $item) {
+          if (isset($item->entity_to_save)) {
+            $file = $item->entity_to_save;
+            $file->save();
+            $file_info = array('target_id' => $file->id());
+
+            $field_definitions = $entity->getFieldDefinitions();
+            $field_type = $field_definitions[$field_name]->getType();
+            // Add alternative text for image type fields.
+            if ($field_type == 'image') {
+              $file_info['alt'] = $file->getFilename();
+            }
+            $entity->{$field_name}[$delta] = $file_info;
+
+            unset($entity->{$field_name}[$delta]->entity_to_save);
+          }
+        }
+      }
```

Comment:
@dickolsson The replication doesn't use this endpoint to POST documents, but we can do this. An example when we use it is when we create the documents in the source database before replication (for testing):

`curl -X POST -H "Content-Type: application/json" -d "$document"  localhost:5984/source`